src/main/java/uk/gov/digital/ho/hocs/domain/repositories/EnumMappingsRepository.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.digital.ho.hocs.domain.repositories;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.domain.repositories.entities.EnumMappings;
+
+import java.util.List;
+
+@Service
+public class EnumMappingsRepository extends JsonConfigFileReader {
+
+    private final List<EnumMappings.EnumMapping> enumMappings;
+
+    public EnumMappingsRepository(ObjectMapper objectMapper) {
+        super(objectMapper);
+
+        enumMappings = readValueFromFile(new TypeReference<>() {});
+    }
+
+    public String getTextValueByNameAndFieldName(String fieldName, String name) {
+        if (!enumMappings.isEmpty()) {
+            for (EnumMappings.EnumMapping mapping : enumMappings) {
+                if (mapping.getFieldName().equals(fieldName)) {
+                    for (EnumMappings.Choice choice : mapping.getChoices()) {
+                        if (choice.getName().equals(name)) {
+                            return choice.getValue();
+                        }
+                    }
+                }
+            }
+        }
+        return "";
+    }
```

Comment:
The level of nesting here and multiple nested for/ifs has a smell. Is there a cleaner and maybe more efficient way of interpreting this and getting the value. Jackson or GSON maybe?

Reply:
Looking back at it, it makes this method clunky doing both 'gets' at once. Could be better for the Repository to just have a getMappingsForFieldName() which can then be iterated through to get the desired mapping